### PR TITLE
Add support for provider network options

### PR DESCRIPTION
**Pull request by Shrews**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

cloud/openstack/os_network.py
##### ANSIBLE VERSION

2.1
##### SUMMARY

Adds new parameters for specifying options for provider network plugins.

### CODE REVIEW

File: cloud/openstack/os_network.py [hunk 1]
```diff
@@ -124,6 +158,10 @@ def main():
         shared=dict(default=False, type='bool'),
         admin_state_up=dict(default=True, type='bool'),
         external=dict(default=False, type='bool'),
+        provider_physical_network=dict(required=False),
```
Comment (rcarrillocruz) [inline cloud/openstack/os_network.py:161]: lacking default here

File: cloud/openstack/os_network.py [hunk 2]
```diff
@@ -124,6 +158,10 @@ def main():
...
+        provider_network_type=dict(required=False, default=None,
+                                   choices=['flat', 'vlan', 'vxlan', 'gre']),
+        provider_segmentation_id=dict(required=False),
```
Comment (rcarrillocruz) [inline cloud/openstack/os_network.py:164]: lacking default

**Comment by rcarrillocruz** [comment]
You lack a couple defaults, but other than that lgtm

**Comment by rcarrillocruz** [comment]
needs_revision

**Comment by rcarrillocruz** [comment]
per chat conversation, non booleans params are ok to not have defaults.

ergo i'm ok to ship it.

**Comment by rcarrillocruz** [comment]
shipit

**Comment by dalees** [comment]
Hi @Shrews - I have verified this change using Ansible 2.0.1.0 and Shade 1.7.0 and used it to create both gre and flat provider networks.

LGTM

**Pull request merged**